test/test_broker.hpp

```diff
@@ -647,7 +647,7 @@ class test_broker {
         do_publish(
             std::move(topic_name),
             std::move(contents),
-            pubopts.get_qos() | pubopts.get_retain(),
+            pubopts,
```

Comment:
Are you sure that you want to forward the duplicate flag as-is?

Reply:
Duplicate flag should be removed. So the code should be `pubopts.get_qos()`.

Reply:
Why wouldn't we forward the retain flag though?

Reply:
Oh I look over it.

It should be

```
pubopts.get_qos() | pubopts.get_retain(),
``` 

I noticed that it is very difficult API for me. I want to remove only dup flag from pubopts. In order to do that, I need to care all other flags. Is there any better way?
Anyway, it is related but different issue.

So I revert the code here.

Reply:
Right now, there is no better way.

One option is to do:

```
pubopts |= dup::no;
publish(..., pubopts, ...);
```